src/main/java/com/samsung/lib/requestserializer/AsyncRequestSerializerConfig.java

```diff
@@ -15,23 +15,60 @@
  */
 package com.samsung.lib.requestserializer;
 
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
 
-
-@Component
 public class AsyncRequestSerializerConfig {
+  final int submitRetryCount;
+  final int submitRetryDelay;
+  final int workerThreadPoolSize;
+  final int localRequestQueueTimeOut;
+
+  private AsyncRequestSerializerConfig(
+      int submitRetryCount,
+      int submitRetryDelay,
+      int workerThreadPoolSize,
+      int localRequestQueueTimeOut
+  ) {
+    this.submitRetryCount = submitRetryCount;
```

Comment:
Actually - we can minimize argument list by passing instance of Builder to this constructor. And initializing like
```java
private AsyncRequestSerializerConfig(Builder builder) {
    this.submitRetryDelay = builder.submitRetryDelay;
    ....
}
```